week3/hackyourweather/src/components/input/Input.jsx

```diff
@@ -0,0 +1,75 @@
+import { useState } from 'react';
+import Button from '../button/Button';
+import CardList from '../card-list/CardList';
+import './Input.css';
+
+const Input = () => {
+  const [city, setCity] = useState(``);
+  const [weatherData, setWeatherData] = useState([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isError, setIsError] = useState({ error: false });
+
+  const handleError = (message) => setIsError({ error: true, message });
+
+  const removeCity = (id, data) => {
+    const newCityList = weatherData.filter((city) => city.id !== id);
+    data ? setWeatherData([data, ...newCityList]) : setWeatherData(newCityList);
+  };
+
+  const showWeather = async () => {
+    const url = `https://api.openweathermap.org/data/2.5/weather?q=${city}&units=metric&appid=${process.env.REACT_APP_OPENWEATHERMAP_API_KEY}
+`;
+
+    try {
+      setIsLoading(true);
+
+      const resp = await fetch(url);
+      if (!resp.ok) throw Error;
+      const data = await resp.json();
+
+      setWeatherData([data]);
+
+      setIsError({ error: false });
+
+      const cityAlreadyListed = weatherData.find((city) => city.id === data.id);
+      cityAlreadyListed
+        ? removeCity(cityAlreadyListed.id, data)
+        : setWeatherData([data, ...weatherData]);
```

Comment:
This feels a bit too long, you can make it a little more readable by using someother part of your code instead of all of this.
```suggestion
      setWeatherData([data, ...weatherData.filter(city => city.id !== data.id)]);
```

It's the same piece you have in `removeCity` function, and I don't feel you need it there either.